api/java/org/openyolo/api/CredentialClient.java

```diff
@@ -141,27 +173,6 @@ public Intent getHintRetrieveIntent(final HintRetrieveRequest request) {
     }
 
     /**
-     * Requests and retrieves any available credentials from the credential providers on the device.
-     * @param request  Properties of the credential request
-     * @param callback Handler for the result of the credential request.
-     */
-    public void retrieve(
-            CredentialRetrieveRequest request,
-            final RetrieveCallback callback) {
```

Comment:
I would lean towards not including it with the rational that:
1) It may cause confusion for more users that it would help.
2) Advanced users playing with the protocol would need to edit the internal callback to change behavior based on responses (changes to request can still be done with the getCredentialRetrieveIntent()).
3) Just a rule of thumb, I think it's best to keep the API as small as possible and add this in the future as needed.